bin/zopen-build

```diff
@@ -1066,52 +1065,84 @@ extractTarBall()
   fi
 }
 
+get_top_folder_name() {
+  archive="$1"
+
+  case "$archive" in
+    *.zip)
+      path=$(unzip -Z1 "$archive" | awk -F/ '{print $1}' | grep -v '^\.$' | sort -u | head -1)
+      ;;
+    *.tar.lz|*.tar.lzip)
+      path=$(lzip -dc "$archive" | tar -tf - 2>/dev/null | awk -F/ '{print $1}' | grep -v '^\.$' | sort -u | head -1)
+      ;;
+    *.tar|*.tar.*|*.tgz)
+      path=$(tar -tf "$archive" 2>/dev/null | awk -F/ '{print $1}' | grep -v '^\.$' | sort -u | head -1)
+      ;;
```

Comment:
Looks good, I think this should be ok in 99% of the cases. @HarithaIBM I think there was one project which had multiple dirs and I'm wondering if we should catch such a situation here as well. Do you recall the project?